### PR TITLE
Django upgrade

### LINKED ISSUE #73: Update Django to latest version
As a developer I want to update Django to it's latest stable version.

### PR DESCRIPTION
**Pull request by froi**
Closes #73 

__TLDR:__ Django was upgraded to 1.10 and so was everything that needed to be upgraded.

Some specifics:

1. Django was updated to latest version as of 2017-03-12
2. The minimum amount of project requirements were upgraded to fit the Django upgrade.
3. South was eliminated along with it's migrations.
4. URL patterns in urls.py files was updated to Django 1.10 formats.
5. settings.py was refactored to conform to Django 1.10
   * Biggest change was to the TEMPLATES_* properties.
6. Added new migrations and eliminated the south migration files.

There's probably more changes to be made. I just thought that we can start looking at these first.

### CODE REVIEW

**Comment by froi** [comment]
TODO:

- [ ] Get Auth providers working.
- [ ] Get Travis checks to green.
